addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -571,11 +572,74 @@ var FieldChar = InputField.extend(TranslatableFieldMixin, {
     tagName: 'span',
     supportedFieldTypes: ['char'],
     isQuickEditable: true,
+    dynamicPlaceholderOptions: {
+        readonly: false,
+        needDefaultValue: true,
+        cancelOnEscape: true,
+        chainedTitle: true,
+        filter: (model) => !["one2many", "boolean", "many2many"].includes(model.type)
+    },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
+
+    /**
+     * @override
+     */
+    init: function () {
+        this._super(...arguments);
+        if (this.nodeOptions && this.nodeOptions.dynamic_placeholder) {
+            // When the dynamic placeholder is active,
+            // the recordData need to be updated when `mailing_model_real` change
+            this.resetOnAnyFieldChange = true;
+        }
+    },
+
+    openDynamicPlaceholder: async function (baseModel, chain = []) {
+        if (!baseModel) {
+            return;
+        }
+        const triggerKeyReplaceRegex = new RegExp(this.nodeOptions.dynamic_placeholder + '$');
+
+        const modelSelector = new ModelFieldSelectorPopOver(this, baseModel, [], this.dynamicPlaceholderOptions);
+        modelSelector.on("field_chain_changed", undefined, (ev) => {
+            this.$el.focus();
+            if (ev.data.chain.length) {
+                let dynamicPlaceholder = "{{object." + ev.data.chain.join('.');
+                const defaultValue = ev.data.defaultValue;
+                dynamicPlaceholder += defaultValue && defaultValue !== '' ? ` or '''${defaultValue}'''}}` : '}}';
+                this.$el[0].value =
```

Comment:
Isn't `this.$el[0]` `this.el` ? If yes then of course it applies to every instance of `this.$el[0]`.

Reply:
indeed, much cleaner :+1: